sos/policies/distros/redhat.py

```diff
@@ -430,11 +433,26 @@ def upload_sftp(self):
                                                        password=_token)
         raise Exception("Could not retrieve valid or anonymous credentials")
 
+    def check_file_too_big(self, archive):
+        size = os.path.getsize(archive)
+        # Lets check if the size is bigger than the limit.
+        # There's really no need to transform the size to Gb,
+        # so we don't need to call any size converter implemented
+        # in tools.py
+        if len(str(size)) >= 10 and (size >= self._max_size_request):
+            self.ui_log.error(
+                _("Size of archive is bigger than Red Hat Customer Portal "
+                  f"limit for uploads of {self._max_size_human_readable} via "
+                  "sos http upload. \n")
+                  )
+            self.upload_url = RH_SFTP_HOST
+
     def upload_archive(self, archive):
         """Override the base upload_archive to provide for automatic failover
         from RHCP failures to the public RH dropbox
         """
         try:
+            self.check_file_too_big(archive)
```

Comment:
This should be gated by checking that we are in fact trying to upload to the RH customer portal - the default is for the policy to set the target, but it's entirely possible for it to not be `RH_API_HOST`.

Reply:
Ack, but it will be good to know about the scenarios where the target is not RH_API_HOST when going through sos/policies/distros/redhat.py